src/libeconf.c

```diff
@@ -304,14 +305,18 @@ void setBoolValue(Key_File *kf, char *group, char *key, char *value) {
 /* --- DESTROY FUNCTIONS --- */
 
 // Free memory allocated by key_file
-void destroy(Key_File key_file) {
-  for (int i = 0; i < key_file.alloc_length; i++) {
-    free(key_file.file_entry[i].group);
-    free(key_file.file_entry[i].key);
-    free(key_file.file_entry[i].value);
+void destroy(Key_File *key_file) {
```

Comment:
This function (and others) needs a prefix, otherwise naming conflicts with other libraries are very likely.